frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -159,9 +159,67 @@ const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
     vmWizardInternalActions[InternalActionType.UpdateVmSettingsField](
       id,
       VMSettingsField.MOUNT_WINDOWS_GUEST_TOOLS,
-      { isHidden: asHidden(!isWindows, VMSettingsField.OPERATING_SYSTEM), value: isWindows },
+      {
+        isHidden: asHidden(!isWindows, VMSettingsField.OPERATING_SYSTEM),
+        value: isWindows,
+      },
     ),
   );
+
+  const relevantOptions = iGetRelevantTemplateSelectors(state, id);
+  const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+  const iTemplate =
+    iCommonTemplates && iGetRelevantTemplate(null, iCommonTemplates, relevantOptions);
+
+  if (os && !iTemplate) {
+    const keepFlavor = Object.assign({}, relevantOptions);
+    Object.assign(keepFlavor, { workload: null });
+    // Check for a valid common-template that contain the chosen Flavor
+    if (iGetRelevantTemplate(null, iCommonTemplates, keepFlavor)) {
+      dispatch(
+        vmWizardInternalActions[InternalActionType.UpdateVmSettingsField](
+          id,
+          VMSettingsField.WORKLOAD_PROFILE,
+          { value: null },
+        ),
+      );
+    } else {
+      const keepWorkload = Object.assign({}, relevantOptions);
+      Object.assign(keepWorkload, { flavor: null });
+      // Check for a valid common-template that contain the chosen Workload
+      if (iGetRelevantTemplate(null, iCommonTemplates, keepWorkload)) {
+        dispatch(
+          vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
+            [VMSettingsField.FLAVOR]: { value: null },
+            [VMSettingsField.MEMORY]: {
```

Comment:
this is done by flavorUpdater. Let's move the flavorUpdater down if we need its functionality later